src/confluent_kafka/src/Admin.c

```diff
@@ -331,92 +331,87 @@ static int Admin_set_replica_assignment (const char *forApi, void *c_obj,
+        Py_ssize_t i = 0;
+        char *name = NULL;
+        char *value = NULL;
+
+        if (!PyList_Check(incremental_configs)) {
+                PyErr_Format(PyExc_TypeError,
+                             "expected list type");
+                goto err;
+        }
 
-        while(PyDict_Next(dict, &pos, &ko, &vo)){
-                const char *k;
-                int config_count = 0;
-                Py_ssize_t i = 0;
+        if ((config_entry_count = (int)PyList_Size(incremental_configs)) < 1) {
+                PyErr_Format(PyExc_ValueError,
+                             "expected non-empty list of ConfigEntry "
+                             "to alter incrementally");
+                goto err;
+        }
+
+        for (i = 0; i < config_entry_count; i++) {
+                PyObject *config_entry;
+                PyObject *incremental_operation;
+                int op, r;
                 rd_kafka_error_t *error;
 
-                if (!(ks = cfl_PyObject_Unistr(ko))) {
-                        PyErr_Format(PyExc_ValueError,
-                                     "expected config name to be unicode "
-                                     "string");
-                       goto err;
-                }
-                k = cfl_PyUnistr_AsUTF8(ks, &ks8);
+                config_entry = PyList_GET_ITEM(incremental_configs, i);
 
-                if (!PyList_Check(vo) || (config_count = (int)PyList_Size(vo)) < 1) {
-                        PyErr_SetString(PyExc_ValueError,
-                                        "expected non-empty list of entry values");
+                r = PyObject_IsInstance(config_entry, ConfigEntry_type);
+                if (r == -1)
+                        goto err; /* Exception raised by IsInstance() */
+                else if (r == 0) {
+                        PyErr_SetString(PyExc_TypeError,
+                                        "Expected list of "
+                                        "ConfigEntry objects");
                         goto err;
                 }
 
-                for (i = 0; i < config_count; i++) {
-                        PyObject *config_op_value, *config_value,
-                                 *op_type;
-                        Py_ssize_t config_value_cnt;
-                        config_op_value = PyList_GET_ITEM(vo, i);
-                        rd_kafka_AlterConfigOpType_t op;
-                        const char *v = NULL;
-
-                        if (!PyList_Check(config_op_value) ||
-                            (config_value_cnt = (int)PyList_Size(config_op_value)) < 2) {
-                                PyErr_SetString(PyExc_ValueError,
-                                                "expected operation type and value");
-                                goto err;
-                        }
+                if (!cfl_PyObject_GetAttr(config_entry, "incremental_operation",
+                                          &incremental_operation, NULL, 1, 0))
```

Comment:
I guess `incremental_operation` is Enum and `op` is the int value for it. Can we name them to suggest that?